src/openfermion/ops/_symbolic_operator.py

```diff
@@ -0,0 +1,301 @@
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+"""SymbolicOperator is a base class for QubitOperator and FermionOperator"""
+import copy
+import itertools
+
+import numpy
+from abc import abstractmethod
+
+
+EQ_TOLERANCE = 1e-12
+
+
+class SymbolicOperatorError(Exception):
+    pass
+
+
+class SymbolicOperator(object):
+    """
+    The base class for QubitOperator and FermionOperator. All methods defined
+    here can be accessed from FermionOperator or QubitOperator objects.
+    This is an abstract class and objects of this type cannot be created,
```

Comment:
Why can't they be created? It seems like being able to create them would be necessary for testing.